File: Qnerator/src/example/test_gen/Transform.cpp
#pragma pack(push, 1)
struct Transform {
    float x;
    float y;
    float z;
    float roll;
    float pitch;
    float yaw;

    Transform(float x, float y, float z, float roll, float pitch, float yaw) : x(x), y(y), z(z), roll(roll), pitch(pitch), yaw(yaw) {}

    std::vector<uint8_t> serialize() const {
        std::vector<uint8_t> buffer;
        buffer.insert(buffer.end(), reinterpret_cast<const uint8_t*>(&x), reinterpret_cast<const uint8_t*>(&x + 1));
        buffer.insert(buffer.end(), reinterpret_cast<const uint8_t*>(&y), reinterpret_cast<const uint8_t*>(&y + 1));
        buffer.insert(buffer.end(), reinterpret_cast<const uint8_t*>(&z), reinterpret_cast<const uint8_t*>(&z + 1));
        buffer.insert(buffer.end(), reinterpret_cast<const uint8_t*>(&roll), reinterpret_cast<const uint8_t*>(&roll + 1));
        buffer.insert(buffer.end(), reinterpret_cast<const uint8_t*>(&pitch), reinterpret_cast<const uint8_t*>(&pitch + 1));
        buffer.insert(buffer.end(), reinterpret_cast<const uint8_t*>(&yaw), reinterpret_cast<const uint8_t*>(&yaw + 1));
        return buffer;
    }

    static Transform deserialize(const std::vector<uint8_t>& buffer) {
        size_t offset = 0;
        float x = *reinterpret_cast<const float*>(buffer.data() + offset);
        offset += sizeof(float);
        float y = *reinterpret_cast<const float*>(buffer.data() + offset);
        offset += sizeof(float);
        float z = *reinterpret_cast<const float*>(buffer.data() + offset);
        offset += sizeof(float);
        float roll = *reinterpret_cast<const float*>(buffer.data() + offset);
        offset += sizeof(float);
        float pitch = *reinterpret_cast<const float*>(buffer.data() + offset);
        offset += sizeof(float);
        float yaw = *reinterpret_cast<const float*>(buffer.data() + offset);
        offset += sizeof(float);
        return Transform(x, y, z, roll, pitch, yaw);
    }
};
#pragma pack(pop)

File: Qnerator/src/example/test_gen/Transform.rs
// 자동 생성된 구조체 및 관련 메서드
        #[repr(C)]
        #[derive(Debug, Clone)]
        pub struct Transform {
            x: f32,
    y: f32,
    z: f32,
    roll: f32,
    pitch: f32,
    yaw: f32,}
    
        impl Transform {
            pub fn new(x: f32, y: f32, z: f32, roll: f32, pitch: f32, yaw: f32) -> Self {
                Self {
                    x,
            y,
            z,
            roll,
            pitch,
            yaw,        }
            }
    
            pub fn serialize(&self) -> Vec<u8> {
                let mut buffer = Vec::new();
                buffer.extend(&self.x.to_le_bytes());
        buffer.extend(&self.y.to_le_bytes());
        buffer.extend(&self.z.to_le_bytes());
        buffer.extend(&self.roll.to_le_bytes());
        buffer.extend(&self.pitch.to_le_bytes());
        buffer.extend(&self.yaw.to_le_bytes());
                buffer
            }
    
            pub fn deserialize(buffer: &[u8]) -> io::Result<Self> {
                let mut offset = 0;
                let mut x_bytes = [0u8; 4];
x_bytes.copy_from_slice(&buffer[offset..offset + 4]);
let x = f32::from_le_bytes(x_bytes);
offset += 4;
        let mut y_bytes = [0u8; 4];
y_bytes.copy_from_slice(&buffer[offset..offset + 4]);
let y = f32::from_le_bytes(y_bytes);
offset += 4;
        let mut z_bytes = [0u8; 4];
z_bytes.copy_from_slice(&buffer[offset..offset + 4]);
let z = f32::from_le_bytes(z_bytes);
offset += 4;
        let mut roll_bytes = [0u8; 4];
roll_bytes.copy_from_slice(&buffer[offset..offset + 4]);
let roll = f32::from_le_bytes(roll_bytes);
offset += 4;
        let mut pitch_bytes = [0u8; 4];
pitch_bytes.copy_from_slice(&buffer[offset..offset + 4]);
let pitch = f32::from_le_bytes(pitch_bytes);
offset += 4;
        let mut yaw_bytes = [0u8; 4];
yaw_bytes.copy_from_slice(&buffer[offset..offset + 4]);
let yaw = f32::from_le_bytes(yaw_bytes);
offset += 4;
                Ok(Self {
                                x,
            y,
            z,
            roll,
            pitch,
            yaw,
                })
            }
        }